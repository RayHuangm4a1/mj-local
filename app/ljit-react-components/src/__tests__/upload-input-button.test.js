import React from 'react';
import { shallow, mount, } from 'enzyme';
import {
	PREFIX_CLASS,
	FILE_LIMIT,
	getLastFileName,
} from '../components/upload-input-button';

describe('Upload Input Button', () => {
	let UploadInputButton;

	beforeEach(() => {
		jest.doMock('antd/lib/upload');
		UploadInputButton = require('../components/upload-input-button').default;
	});
	afterEach(() => {
		jest.unmock('antd/lib/upload');
		UploadInputButton = undefined;
	});

	it('should PREFIX_CLASS to equal ljit-upload-input-button', () => {
		expect(PREFIX_CLASS).toEqual('ljit-upload-input-button');
	});

	it('should FILE_LIMIT to equal 1', () => {
		expect(FILE_LIMIT).toEqual(1);
	});

	it('handle default props', () => {
		const {
			text,
			onChange,
		} = UploadInputButton.defaultProps;

		expect(text).toEqual('上传档案');
		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
	});

	it('should renders correctly', () => {
		const wrapper = shallow(<UploadInputButton />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const acceptExtentions = ['txt'];
		const text = 'mock-text';
		const remindText = 'mock-remindtext';
		const onChange = () => {};
		const className = 'mock-calss';
		const fileList = [];
		const wrapper = mount(
			<UploadInputButton
				acceptExtentions={acceptExtentions}
				text={text}
				remindText={remindText}
				onChange={onChange}
				className={className}
				fileList={fileList}
			/>
		);

		expect(wrapper.props().acceptExtentions).toEqual(acceptExtentions);
		expect(wrapper.props().text).toEqual(text);
		expect(wrapper.props().remindText).toEqual(remindText);
		expect(wrapper.props().onChange).toEqual(onChange);
		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().fileList).toEqual(fileList);
	});

	describe('getLastFileName', () => {
		it('should equal file1, when pass [{ name: \'file1\' }]', () => {
			const fileList = [{ name: 'file1', },];
			const actual = 'file1';

			expect(getLastFileName(fileList)).toEqual(actual);
		});

		it('should equal file2, when pass [{ name: \'file1\' }, { name: \'file2\' }]', () => {
			const fileList = [
				{ name: 'file1', },
				{ name: 'file2', },
			];
			const actual = 'file2';

			expect(getLastFileName(fileList)).toEqual(actual);
		});

		it('should equal \'\', when pass [{}]', () => {
			const fileList = [{}];
			const actual = '';

			expect(getLastFileName(fileList)).toEqual(actual);
		});

		it('should equal \'\', when pass []', () => {
			const fileList = [];
			const actual = '';

			expect(getLastFileName(fileList)).toEqual(actual);
		});
	});
});
