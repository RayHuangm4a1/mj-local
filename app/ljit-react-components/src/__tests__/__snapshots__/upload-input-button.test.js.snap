// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Input Button should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UploadInputButton
    onChange={[Function]}
    text="上传档案"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "acceptExtentions": Array [],
      "beforeUpload": [Function],
      "children": Array [
        <Input
          className="ljit-upload-input-button__input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onPressEnter={[Function]}
          readOnly={true}
          value=""
        />,
        <Button
          block={false}
          className="ljit-upload-input-button__button"
          ghost={false}
          htmlType="button"
          loading={false}
        >
          <Icon
            type="upload"
          />
          上传档案
        </Button>,
      ],
      "className": "ljit-upload-input-button",
      "fileLimit": 1,
      "fileList": undefined,
      "isUploadListVisible": false,
      "listType": "text",
      "onChange": [Function],
      "onRemove": [Function],
      "remindText": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "ljit-upload-input-button__input",
          "disabled": false,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "onPressEnter": [Function],
          "readOnly": true,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "children": Array [
            <Icon
              type="upload"
            />,
            "上传档案",
          ],
          "className": "ljit-upload-input-button__button",
          "ghost": false,
          "htmlType": "button",
          "loading": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "type": "upload",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "上传档案",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "acceptExtentions": Array [],
        "beforeUpload": [Function],
        "children": Array [
          <Input
            className="ljit-upload-input-button__input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onPressEnter={[Function]}
            readOnly={true}
            value=""
          />,
          <Button
            block={false}
            className="ljit-upload-input-button__button"
            ghost={false}
            htmlType="button"
            loading={false}
          >
            <Icon
              type="upload"
            />
            上传档案
          </Button>,
        ],
        "className": "ljit-upload-input-button",
        "fileLimit": 1,
        "fileList": undefined,
        "isUploadListVisible": false,
        "listType": "text",
        "onChange": [Function],
        "onRemove": [Function],
        "remindText": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "ljit-upload-input-button__input",
            "disabled": false,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "onPressEnter": [Function],
            "readOnly": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": Array [
              <Icon
                type="upload"
              />,
              "上传档案",
            ],
            "className": "ljit-upload-input-button__button",
            "ghost": false,
            "htmlType": "button",
            "loading": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "type": "upload",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "上传档案",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
